packages/headless/src/controllers/core/triggers/headless-core-notify-trigger.test.ts

```diff
@@ -0,0 +1,153 @@
+import {logNotifyTrigger} from '../../../features/triggers/trigger-analytics-actions';
+import {triggerReducer as triggers} from '../../../features/triggers/triggers-slice';
+import {
+  buildMockSearchEngine,
+  MockedSearchEngine,
+} from '../../../test/mock-engine-v2';
+import {createMockState} from '../../../test/mock-state';
+import {NotifyTrigger} from '../../core/triggers/headless-core-notify-trigger';
+import {buildCoreNotifyTrigger} from './headless-core-notify-trigger';
+
+jest.mock('../../../features/triggers/trigger-analytics-actions');
+
+describe('NotifyTrigger', () => {
+  let engine: MockedSearchEngine;
+  let notifyTrigger: NotifyTrigger;
+
+  function initNotifyTrigger() {
+    notifyTrigger = buildCoreNotifyTrigger(engine, {
+      options: {
+        logNotifyTriggerActionCreator: logNotifyTrigger,
+      },
+    });
+  }
+
+  function setEngineTriggersState(notifications: string[]) {
+    engine.state.triggers!.notifications = notifications;
+  }
+
+  function registeredListeners() {
+    return (engine.subscribe as jest.Mock).mock.calls.map((args) => args[0]);
+  }
+
+  beforeEach(() => {
+    jest.resetAllMocks();
```

Comment:
We're now using vitest instead of jest

The APIs are very similar, so the migration is straightforward (e.g., here I think you'd just have to do `vi.resetAllMocks();`)